src/xenia/base/system_gnulinux.cc

```diff
@@ -29,7 +29,11 @@ void LaunchWebBrowser(const std::string_view url) {
   system(cmd.c_str());
 }
 
-void LaunchFileExplorer(const std::filesystem::path& path) { assert_always(); }
+void LaunchFileExplorer(const std::filesystem::path& path) {
+  auto cmd = std::string("xdg-open ");
+  cmd.append(path);
```

Comment:
All shell-interpreted characters, as well as spaces, possibly need to be escaped if they are actually interpreted if `system` is used — this applies to launching the web browser too.

Reply:
I think ideally this shouldn't go through the shell at all — is `posix_spawn` the right way of doing the same, but with explicit `argv`?

Reply:
Because the input to `system`, `xdg-open`, the content directory path and specific urls is hard-coded essentially, and all of those work correctly there isn't really a need to escape anything, Unless I'm missing something?

Regarding using `system` or `posix_spawn`, `posix_spawn` would probably be the "better" way of doing it in that sense from what I have seen, I haven't used `posix_spawn` before though and am currently unable to get it to compile without complaining about converting the `path` and `xdg-open` strings to `char *` when it comes to `argv`. I will continue looking into this as it's probably just something I'm not quite getting right now.

Reply:
The content directory path isn't hardcoded, it's by default relative to the user's home directory, but can be changed arbitrarily in the config.

Reply:
Right, Makes sense. I have put together a commit to replace `system()` with `posix_spawn`. What do you think? I'll do the same with `LaunchWebBrowser` if your ok with this one.

Reply:
If it works fine, and replicates the normal behavior of launching from `system` (such as, all environment variables are properly passed), that would be awesome.

Reply:
Yep, The users environment variables are pulled in using the `environ` variable so xdg-open works correctly. Works the same as `system()` from the users perspective.